src/sentry/options/manager.py

```diff
@@ -476,7 +476,7 @@ def can_update(self, key: str, value, channel: UpdateChannel) -> NotWritableReas
         opt = self.lookup_key(key)
         if opt.has_any_flag({FLAG_NOSTORE, FLAG_IMMUTABLE}):
             return NotWritableReason.READONLY
-        if opt.has_any_flag({FLAG_PRIORITIZE_DISK}) and settings.SENTRY_OPTIONS.get(key):
+        if opt.has_any_flag({FLAG_PRIORITIZE_DISK}) and key in settings.SENTRY_OPTIONS:
```

Comment:
I also tried `opt.default() != settings.SENTRY_OPTIONS.get(key)` but if the key doesn't exist it will return None. 
None != most default option values (like `0`, or `""`) 
resulting in an incorrect evaluation that the option is set on disk.